packages/native-core/src/layout_attributes.rs

```diff
@@ -19,573 +19,400 @@
+                    bottom: convert_length_percentage_or_auto(padding.bottom),
+                    left: convert_length_percentage_or_auto(padding.left),
+                    right: convert_length_percentage_or_auto(padding.right),
+                };
+            }
+            Property::Width(width) => {
+                style.size.width = convert_size(width);
+            }
+            Property::Height(height) => {
+                style.size.height = convert_size(height);
+            }
+            _ => (),
+        }
+        // currently not implemented in lightningcss
+        if name == "aspect-ratio" {
+            if let Ok(ratio) = Ratio::parse_string(value) {
+                style.aspect_ratio = Some(ratio.0 / ratio.1);
             }
         }
-        _ => (),
     }
 }
 
-fn apply_animation(name: &str, _value: &str, _style: &mut Style) {
-    match name {
-        "animation" => {}
-        "animation-delay" => {}
-        "animation-direction =>{}" => {}
-        "animation-duration" => {}
-        "animation-fill-mode" => {}
-        "animation-itera =>{}tion-count" => {}
-        "animation-name" => {}
-        "animation-play-state" => {}
-        "animation-timing-function" => {}
-        _ => {}
+fn convert_length_value(length_value: LengthValue) -> Dimension {
+    match length_value {
+        LengthValue::Px(value) => Dimension::Points(value),
+        _ => todo!(),
     }
 }
 
-fn apply_column(name: &str, _value: &str, _style: &mut Style) {
-    match name {
-        "column-count" => {}
-        "column-fill" => {}
-        "column-gap" => {}
-        "column-rule" => {}
-        "column-rule-color" => {}
-        "column-rule-style" => {}
-        "column-rule-width" => {}
-        "column-span" => {}
-        "column-width" => {}
-        _ => {}
+fn convert_dimention_percentage(
+    dimension_percentage: DimensionPercentage<LengthValue>,
+) -> Dimension {
+    match dimension_percentage {
+        DimensionPercentage::Dimension(value) => convert_length_value(value),
+        DimensionPercentage::Percentage(percentage) => Dimension::Percent(percentage.0),
+        _ => todo!(),
     }
 }
 
-fn apply_flex(name: &str, value: &str, style: &mut Style) {
-    // - [x] pub flex_direction: FlexDirection,
-    // - [x] pub flex_wrap: FlexWrap,
-    // - [x] pub flex_grow: f32,
-    // - [x] pub flex_shrink: f32,
-    // - [x] pub flex_basis: Dimension,
-
-    match name {
-        "flex" => {}
-        "flex-direction" => {
-            use FlexDirection::*;
-            style.flex_direction = match value {
-                "row" => Row,
-                "row-reverse" => RowReverse,
-                "column" => Column,
-                "column-reverse" => ColumnReverse,
-                _ => Row,
-            };
-        }
-        "flex-basis" => {
-            if let Some(v) = parse_value(value) {
-                style.flex_basis = v;
-            }
+fn convert_length_percentage_or_auto(
```

Comment:
Doesn't change anything for now, but just FYI, Taffy 0.3 has `LengthPercentage` and `LengthPercentageAuto` types in addition to `Dimension`.